soc/arm/microchip_mec/mec1501/device_power.c

```diff
@@ -133,33 +137,47 @@ const struct ds_timer_info ds_timer_tbl[] = {
 		(MCHP_CCT_CTRL_COMP1_SET | MCHP_CCT_CTRL_COMP0_SET),
 	},
 };
-#define NUM_DS_TIMER_ENTRIES \
-	(sizeof(ds_timer_tbl) / sizeof(struct ds_timer_info))
 
+static struct ds_dev_info ds_ctx;
 
-static uint32_t timers[NUM_DS_TIMER_ENTRIES];
-static uint8_t uart_activate[3];
+static void deep_sleep_save_ecs(void)
+{
+	ds_ctx.ecs[0] = ECS_REGS->ETM_CTRL;
+	ds_ctx.ecs[1] = ECS_REGS->DEBUG_CTRL;
+#ifdef CONFIG_DEEP_SLEEP_JTAG
+	ECS_REGS->ETM_CTRL = 0;
+	ECS_REGS->DEBUG_CTRL = 0x00;
+#endif
+}
 
 static void deep_sleep_save_uarts(void)
 {
-	uart_activate[0] = UART0_REGS->ACTV;
-	if (uart_activate[0]) {
+	ds_ctx.uart_info[0] = UART0_REGS->ACTV;
+	if (ds_ctx.uart_info[0]) {
 		while ((UART0_REGS->LSR & MCHP_UART_LSR_TEMT) == 0) {
 		};
 	}
 	UART0_REGS->ACTV = 0;
-	uart_activate[1] = UART1_REGS->ACTV;
-	if (uart_activate[1]) {
+	ds_ctx.uart_info[1] = UART1_REGS->ACTV;
+	if (ds_ctx.uart_info[1]) {
 		while ((UART1_REGS->LSR & MCHP_UART_LSR_TEMT) == 0) {
 		};
 	}
 	UART1_REGS->ACTV = 0;
-	uart_activate[2] = UART2_REGS->ACTV;
-	if (uart_activate[2]) {
+	ds_ctx.uart_info[2] = UART2_REGS->ACTV;
+	if (ds_ctx.uart_info[2]) {
 		while ((UART2_REGS->LSR & MCHP_UART_LSR_TEMT) == 0) {
 		};
 	}
+
+	/* TODO: Explore only allow to sleep but not deactivate */
+	mchp_pcr_periph_slp_ctrl(PCR_UART0, MCHP_PCR_SLEEP_EN);
```

Comment:
@scottwcpg does enabling sleep for individual peripherals at this point has any effect in how quick we enter SoC LPM?

Is it preferred to keep completely deactivated the peripheral or would enable the sleep be enough to guarantee CLK_REQ doesn't block LPM?

Reply:
Peripherals require PCR SLP_EN bit = 1 so the chip's PCR HW will disable the clock tree going to the peripheral when CLK_REQ goes to 0.  UART is a problem because it's an old HW design and does not follow the PCR SLP_EN rules. Setting the UART's logical device register = 1 causes the UART's CLK_REQ -> 1 even if the UART is idle.  The only way to cause the UART CLK_REQ -> 0 is to clear the UART logical device Activate register.

Reply:
Got it @scottwcpg  I have removed the code since won't make any difference